numpy/lib/arraypad.py

```diff
@@ -1052,7 +1052,7 @@ def _normalize_shape(ndarray, shape, cast_to_int=True):
             arr = arr.repeat(2, axis=1)
         elif arr.shape[0] == ndims:
             # Input correctly formatted, pass it on as `arr`
-            arr = shape
+            arr = np.asarray(shape)
```

Comment:
I think this would clearer written as
```
# Input already correct; nothing to do
pass
```
(note that `arr` was set to `np.asarray(shape)` above -- as written this line just resets it to what it already is)